SchemaValidator/SchemaValidation.py

```diff
@@ -0,0 +1,218 @@
+              + '\tvalidationFeedback:' + str(validationFeedback))
+        return False
+    if(url is None):
+        url = getUrl()
+    if(token is None):
+        token = getToken(url=url)
+
+    uploadID = getUploadID(url=url, uploadFileName=uploadFileName, token=token)
+    validationSuccess=updateValidationStatus(uploadID=uploadID, isValid=isValid, validationFeedback=validationFeedback, url=url, token=token)
+
+    return validationSuccess
+
+def validateAgainstSchema(schemaText=None, xmlText=None):
+    try:
+        # Validate XML against Schema.
+        schemaTree = etree.XMLSchema(etree.XML(schemaText))
+        xmlParser = etree.XMLParser(schema=schemaTree)
+        try:
+            xmlTree = etree.fromstring(xmlText, xmlParser)
+            # If we get this far, the XML has validated successfully.
+            return ('Valid')
+        except etree.XMLSyntaxError as err:
+            return ('Invalid:\t' + str(err))
+
+    except Exception as e:
+        print('Unexpected problem during xml validation.')
+        print(str(e))
+        print(exc_info())
+        return (str(e) + '\n' + str(exc_info()))
+
+def updateValidationStatus(uploadID=None, isValid=None, validationFeedback=None, token=None, url=None):
+    if(url is None):
+        url=getUrl()
+    try:
+        fullUrl = str(url) + '/api/uploads/' + str(uploadID) + '/'
+        # TODO: Haml Files?
+        body = {'valid': isValid, 'validationFeedback': validationFeedback, 'id':uploadID, 'type':'HML'}
+        encodedJsonData = str(json.dumps(body)).encode('utf-8')
+        updateRequest = request.Request(url=fullUrl, data=encodedJsonData, method='PUT')
+        updateRequest.add_header('Content-Type', 'application/json')
+        updateRequest.add_header('Authorization', 'Bearer ' + token)
+        responseData = request.urlopen(updateRequest).read().decode("UTF-8")
+        response=json.loads(responseData)
+
+        # The response contains the saved data, if successful. If the response matches what we expect it was a success.
+        if(str(response['id'])==str(uploadID) and str(response['valid'])==str(isValid)
+            and str(response['validationFeedback'])==str(validationFeedback)):
+            return True
+        else:
+            print('Could not set validation status, response:\n' + str(response))
+            return False
+    except SyntaxError as e:
+        print('Syntax error when parsing response from request:\n' + str(e) + '\n' + str(exc_info()))
+        return False
+    except urllib.error.HTTPError as e:
+        print('HTTP error when setting validation status for upload id ' + str(uploadID) + ' : ' + str(e))
+        return False
+    except Exception as e:
+        print('Error when updating validation status:\n' + str(e) + '\n' + str(exc_info()))
+        return False
+
+def getUploadID(uploadFileName=None, token=None, url=None):
+    print('Trying to find upload ID for file name:' + str(uploadFileName))
```

Comment:
I appreciate your effort, but we could have also solved it by adding the ID to the file name ;-)

Reply:
Do we want to keep asking for all uploads? Not sure this scales well.

Reply:
I agree it's not great. I'm not yet including the file ID in the filename, but the file ID is found in the JAVA code within the REST method. It's better this way, but long term maybe we can implement a uploadRepository.findByFilename() method, unless it's already there..